pkg/clusteragent/admission/mutate/autoinstrumentation/auto_instrumentation.go

```diff
@@ -648,7 +649,12 @@ func (w *Webhook) injectAutoInstruConfig(pod *corev1.Pod, libsToInject []libInfo
 				{
 					key:     dotnetProfilingLdPreloadKey,
 					valFunc: dotnetProfilingLdPreloadEnvValFunc,
-				}})
+					isEligibleToInject: func(c corev1.Container) bool {
+						idx := mutatecommon.EnvIndex(c.Env, "DD_PROFILING_ENABLED")
```

Comment:
nit: Not directly related to this PR but we can use the stdlib slices.indexFunc instead of defining our own EnvIndex.
https://pkg.go.dev/slices#IndexFunc

Reply:
What version of Go are we using to build the agent? I agree that we should use as much stdlib stuff as possible, but the slices package is pretty new, don't know what our agent Go version policy is.